src/main/java/fr/insee/genesis/domain/service/ScheduleImpl.java

```diff
@@ -0,0 +1,77 @@
+package fr.insee.genesis.domain.service;
+
+import fr.insee.genesis.domain.ports.api.ScheduleApiPort;
+import fr.insee.genesis.exceptions.InvalidCronExpressionException;
+import fr.insee.genesis.exceptions.NotFoundException;
+import fr.insee.genesis.infrastructure.model.document.schedule.KraftwerkExecutionSchedule;
+import fr.insee.genesis.infrastructure.model.document.schedule.StoredSurveySchedule;
+import fr.insee.genesis.infrastructure.model.document.schedule.ServiceToCall;
+import fr.insee.genesis.infrastructure.repository.ScheduleMongoDBRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.support.CronExpression;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@Slf4j
+public class ScheduleImpl implements ScheduleApiPort {
+    private final ScheduleMongoDBRepository scheduleMongoDBRepository;
+
+    @Autowired
+    public ScheduleImpl(ScheduleMongoDBRepository scheduleMongoDBRepository) {
+        this.scheduleMongoDBRepository = scheduleMongoDBRepository;
+    }
+
+    @Override
+    public List<StoredSurveySchedule> getAllSchedules() {
+        return scheduleMongoDBRepository.findAll();
+    }
+
+    @Override
+    public void addSchedule(String surveyName, ServiceToCall serviceToCall, String frequency, LocalDateTime scheduleBeginDate, LocalDateTime scheduleEndDate) throws InvalidCronExpressionException{
+        //Fequency format check
+        if(!CronExpression.isValidExpression(frequency)) {
+            throw new InvalidCronExpressionException();
+        }
+        
+        List<StoredSurveySchedule> storedSurveySchedules = scheduleMongoDBRepository.findBySurveyName(surveyName);
+
+        StoredSurveySchedule storedSurveySchedule;
+        if (storedSurveySchedules.isEmpty()) {
+            //Create if not exists
+            log.info("Creation of new survey document for survey " + surveyName);
+            storedSurveySchedules.add(new StoredSurveySchedule(surveyName, new ArrayList<>()));
+        }
+        ScheduleUnicityService scheduleUnicityService = new ScheduleUnicityService();
+        storedSurveySchedule = scheduleUnicityService.deduplicateSurveySchedules(surveyName, storedSurveySchedules);
```

Comment:
If there is more than one StoredSurveySchedule for one survey (which is not what we want), we probably should delete these from database before saving the new deduplicated one.